src/SPC/builder/unix/library/nghttp3.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SPC\builder\unix\library;
+
+use SPC\exception\FileSystemException;
+use SPC\exception\RuntimeException;
+use SPC\exception\WrongUsageException;
+
+trait nghttp3
+{
+    /**
+     * @throws FileSystemException
+     * @throws RuntimeException
+     * @throws WrongUsageException
+     */
+    protected function build(): void
+    {
+        $args = $this->builder->makeAutoconfArgs(static::NAME, [
+            'zlib' => null,
+            'openssl' => null,
+        ]);
+
+        shell()->cd($this->source_dir)
+            ->setEnv([
+                'CFLAGS' => $this->getLibExtraCFlags(),
+                'LDFLAGS' => $this->getLibExtraLdFlags(),
+                'LIBS' => $this->getLibExtraLibs(),
+            ])
+            ->execWithEnv(
+                './configure ' .
+                '--enable-static ' .
+                '--disable-shared ' .
+                '--with-pic ' .
+                '--enable-lib-only ' .
+                $args . ' ' .
+                '--prefix='
+            )
+            ->execWithEnv('make clean')
+            ->execWithEnv("make -j{$this->builder->concurrency}")
+            ->execWithEnv('make install DESTDIR=' . BUILD_ROOT_PATH);
+        $this->patchPkgconfPrefix(['libnghttp3.pc']);
+    }
```

Comment:
apart from the available AutoconfArgs, all three libraries use the exact same code to build

Maybe we should start thinking about defining the method once somewhere and only passing the available options to it.